### PR TITLE
L&F UI Update for Backend Changes

### PR DESCRIPTION
**Pull request by richmondnao**
Added code for Data entry entering a missing item for someone else, also added new changes to meet the criteria of the new database and a bunch of minor things to other pages just to fit with the new database

### CODE REVIEW

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx
Comment (ben-1010) [inline src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx:274]: minRows={8} to fill right column of form

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx
Comment (ben-1010) [inline src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx:290]: minRows={8} to fill right column of form

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx
Comment (ben-1010) [inline src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx:59]: Is this component being used anywhere?

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ItemCategory/index.tsx
Comment (ben-1010) [inline src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ItemCategory/index.tsx:47]: Similarly, is this component being used?

File: src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx
Comment (ben-1010) [inline src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx:230]: This is a strange visual progression...
Screenshot 2024-11-12 at 2 41 08 PM
Screenshot 2024-11-12 at 2 41 43 PM
Screenshot 2024-11-12 at 2 42 12 PM
@
Screenshot 2024-11-12 at 2 42 39 PM
Screenshot 2024-11-12 at 2 43 21 PM

**Review by ben-1010 — COMMENTED** [review]
Overall looks pretty good, added some comments of things that could be improved

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx:59]: Currently no, I'm still working on how to incorporate that, i might remove it in my next iteration of what i work on if im still not able to figure that out

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ItemCategory/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ItemCategory/index.tsx:47]: Same response

File: src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx:230]: Not sure what you mean ?

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/components/confirmReport/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/components/confirmReport/index.tsx:77]: I think its because the report stolen button sends the stores the stolen description data once the person clicks on report stolen, i'll have to figure out a way to work around that. But honestly, i'm not sure why someone would report something stolen and right then go back to change that from being stolen, but yh its something that i guess needs to work better

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx:284]: why 8?

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx:300]: why 8? seems fine on my end, is there a specific reason?

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx
Comment (Centerville1) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx:59]: If it's unused, it should be removed from this PR before we merge, you can keep it locally if it's in progress, but I don't think we shouldn't merge it to our shared branch.

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ItemCategory/index.tsx
Comment (Centerville1) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ItemCategory/index.tsx:47]: See above comment @richmondnao

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx:59]: sounds good

File: src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx
Comment (Centerville1) [reply src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx:230]: I believe he means the filter section looks very weird when window size shrinks, particularly the new "report item for others" button.  That button should probably move either on top of or under the filters on mobile, to fit better.

Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx:230]: it does move to the bottom for a specific size screen
Screenshot 2024-11-12 at 9 35 27 PM

Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx:230]: its more complicated adjusting each of them for each screen size, this seemed like the best solution, that way it also doesnt affect the data view for the items

Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx:230]: Its on the side on screen that have room on the grid for it to fit in there and the screen that dont it adjust accordingly, untill its all on one screen size, so each of them keep shrinking kind of

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/ColorComponent/index.tsx]: > @richmondnao If it's unused, it should probably be removed from this PR before we merge, you can keep it in a branch locally, or in a separate PR if it's in progress, but I don't think we shouldn't merge it to our shared branch.

removed unused components

File: src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx
Comment (Centerville1) [reply src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/index.tsx:230]: Currently it jumps around a lot as the screen changes size.  It's pretty jarring, so if we can work on the breakpoints to be a little more consistent (like all jump at once), that would be good.  Not a huge deal, but should be improved at some point.


https://github.com/user-attachments/assets/a24ca602-522d-43ae-b3bc-d0a177d53284

**Review by Centerville1 — CHANGES_REQUESTED** [review]
Overall pretty good, but there's still a bit that needs work, and some bugs that need ironing out.  I also left some questions on some places I wasn't sure about, but good work in general!

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx
Comment (Centerville1) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx:155]: Also go through, search, and delete other console logs, in case I missed some.

File: src/views/CampusSafety/index.tsx
Comment (Centerville1) [reply src/views/CampusSafety/index.tsx:25]: Or maybe not, but have a button to get back to the list page from the report page (as per the Figma design).

File: src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/components/MissingItemReportData/ReportItemPageOther/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFoundAdmin/views/MissingItemList/components/MissingItemReportData/ReportItemPageOther/index.tsx:1]: I thought i had that in the components folder because it shows up on the MissingItemList pge, so it made sense to put it under the components of that folder, i can move it to the components folder i think that makes sense

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/confirmReport/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemEdit/components/confirmReport/index.tsx:73]: No sure what you mean here?

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx:68]: Well i removed it because i thought i was done making changes to them, but then realized that I might need them again, that was on me

File: src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx
Comment (richmondnao) [reply src/views/CampusSafety/views/LostAndFound/views/MissingItemCreate/index.tsx:78]: I believe it was because i changed how the handlechange was being handled and didnt need the part to format the date, since its already being done on the confirmation page.

File: src/services/lostAndFound.ts
Comment (Centerville1) [inline src/services/lostAndFound.ts:1]: the...

**Comment by Centerville1** [comment]
LGTM

**Review by Centerville1 — APPROVED** [review]
LGTM

**Pull request merged**
